crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4180,6 +4224,108 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn except_handler_single_exception() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_dedented(
+            "src/a.py",
+            "
+            import re
+
+            try:
+                x
+            except NameError as e:
+                pass
+            except re.error as f:
+                pass
+            ",
+        )?;
+
+        assert_public_ty(&db, "src/a.py", "e", "NameError");
+        assert_public_ty(&db, "src/a.py", "f", "error");
```

Comment:
Not entirely happy that the display of `Type::Instance(re.error)` here is `error`, but that's definitely out of scope for this issue